backend/redis/queue.go

```diff
@@ -45,16 +45,25 @@ type KeyInfo struct {
 }
 
 func newTaskQueue[T any](ctx context.Context, rdb redis.UniversalClient, keyPrefix string, tasktype string) (*taskQueue[T], error) {
+	return newTaskQueueWithWorkerName[T](ctx, rdb, keyPrefix, tasktype, "")
+}
+
+func newTaskQueueWithWorkerName[T any](ctx context.Context, rdb redis.UniversalClient, keyPrefix string, tasktype string, workerName string) (*taskQueue[T], error) {
```

Comment:
Let's merge the two and add a new parameter to `newTaskQueue` for the worker name